atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -314,15 +314,15 @@ private int getKnownPort() throws UnknownHostException {
         }
     }
 
-    private Set<InetSocketAddress> getAllKnownHosts() {
-        return ImmutableSet.copyOf(Sets.union(getProxiesFromCurrentPool(), getServersSocketAddressesFromConfig()));
+    private Sets.SetView<InetSocketAddress> getAllKnownHosts() {
```

Comment:
this isn't wrong, but do we gain much from not using `Set` as the return type here? I suppose the intent is to make it clear that this is a view that may change underneath you?